### PR TITLE
Add NSG and NSG Rule support

### PR DESCRIPTION
**Pull request by matthchr**
**What this PR does / why we need it**:
Adds support for NetworkSecurityGroup and NetworkSecurityGroupRules

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: hack/generated/config/samples/microsoft.network/v1alpha120201101_networksecuritygroup.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: microsoft.network.infra.azure.com/v1alpha1api20201101
```
Comment (matthchr) [inline hack/generated/config/samples/microsoft.network/v1alpha120201101_networksecuritygroup.yaml:1]: This isn't `infra` anymore. Need to fix

Comment (matthchr) [reply hack/generated/config/samples/microsoft.network/v1alpha120201101_networksecuritygroup.yaml:1]: same below

File: hack/generated/controllers/crd_networking_nsg_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			SourceAddressPrefix:      to.StringPtr("*"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			Access:                   network.SecurityRulePropertiesFormatAccessDeny,
+			Priority:                 124,
+			Direction:                network.SecurityRulePropertiesFormatDirectionInbound,
+		},
+	}
+
+	tc.CreateResourcesAndWait(rule1, rule2)
+	defer tc.DeleteResourceAndWait(rule1)
```
Comment (Porges) [inline hack/generated/controllers/crd_networking_nsg_test.go:112]: if we have a `CreateResources` we should have a `DeleteResources`

**Review by Porges — APPROVED** [review]

File: hack/generated/controllers/crd_networking_nsg_test.go (hunk 2 above)
Comment (matthchr) [reply hack/generated/controllers/crd_networking_nsg_test.go:112]: Fixed

**Pull request merged**